### PR TITLE
#11 graphql mutations

### LINKED ISSUE #11: Basic graphql mutations
Have the read operations, now need the create, read and delete operations!

Basic mutations:

Create a requisition, add requisition lines, delete lines/requisitions. 

Not sure if we should bother with the operations for items/stores - probs not!

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #11 

Just some simple mutations. Decided to just leave deletes - f it!?

i.e.

```
mutation{
  addRequisition(storeId: "store1"){
    store{
      name
    }
    lines{
      quantity
    }
  }
}
```

adds a new requisition!

### CODE REVIEW

File: src/schema/Mutation.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  type Mutation {
+    addRequisitionLine(requisitionId: String, itemId: String): RequisitionLine
+    addRequisition(storeId: String): Requisition
+    updateRequisitionLineQuantity(id: String, quantity: Int): RequisitionLine
+  }
+`;
+
+export const MutationResolver = {
+  Mutation: {
+    addRequisition: (_: any, args: any) => {
```
Comment (wlthomson) [inline src/schema/Mutation.ts:15]: Initialises a new requisition?

File: src/schema/Mutation.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        id: `req${Math.floor(Math.random() * 1000)}`,
+        store: storeId,
+        lines: [],
+      };
+      store.requisitions.push(newRequisition.id);
+      requisitions.push(newRequisition);
+
+      return newRequisition;
+    },
+    addRequisitionLine: (_: any, args: any) => {
```
Comment (wlthomson) [inline src/schema/Mutation.ts:29]: Adds a new requisition line to an existing requisition?

File: src/schema/Mutation.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        id: `reqLine${Math.floor(Math.random() * 1000)}`,
+        quantity: 0,
+        item: itemId,
+      };
+      currentRequisition.lines.push(newRequisitionLine.id);
+      requisitionLines.push(newRequisitionLine);
+
+      return newRequisitionLine;
+    },
+    updateRequisitionLineQuantity: (_: any, args: any) => {
```
Comment (wlthomson) [inline src/schema/Mutation.ts:45]: Updates quantity of a requisition line for an existing requisition?

**Comment by joshxg** [comment]
@wlthomson yup commnets are correct - will add some api docs in a tidy up - obv this code is a bit ugo at the mo 🤣

**Pull request merged**